### PR TITLE
Add icon to url text box and enable open valid url

### PR DESCRIPTION
**Pull request by mayayunx**
Signed-off-by: Ruiyun Xie <ruiyun.xie@tum.de>

### Summary of changes

- Add icon to the URL Textfield for opening a link in browser
- Check if the url is empty, then disable the icon and change the color
- Check if url is valid, otherwise add https protocol to it
- Open link via clicking on the icon if not disabled 

### Context and reason for change

Enable the user open link in browser.

### How can the changes be tested

By entering different links and clicking on the icon.

Note: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

### CODE REVIEW

File: src/Frontend/Components/AttributionColumn/PackageSubPanel.tsx [hunk 1]
```diff
@@ -6,9 +6,19 @@
...
 import { useAttributionColumnStyles } from './shared-attribution-column-styles';
+import OpenInNewIcon from '@material-ui/icons/OpenInNew';
+import { IconButton } from '../IconButton/IconButton';
+import { makeStyles } from '@material-ui/styles';
+import { clickableIcon, disabledIcon } from '../../shared-styles';
+
+const useStyles = makeStyles({
+  clickableIcon: clickableIcon,
+  disabledIcon: disabledIcon,
+});
```
Comment (leslielazzarino) [inline src/Frontend/Components/AttributionColumn/PackageSubPanel.tsx:21]: Can be written as: `const useStyles = makeStyles({clickableIcon,disabledIcon});`

**Review by leslielazzarino — CHANGES_REQUESTED** [review]
- The only URL in the opossum_input.json example file has a typo (typescRipt), preventing to directly test this feature on that file
- The icon on the path bar has moved to the left from its expected positon at the extreme right.

**Review by benedikt-richter — APPROVED** [review]
consider squashing individual commits before merging

File: src/Frontend/Components/GoToLinkButton/GoToLinkButton.tsx [hunk 2]
```diff
@@ -74,13 +77,23 @@ export function GoToLinkButton(props: GoToLinkProps): ReactElement {
...
+      tooltipTitle={
+        isLocalLink(openLinkArgs.link)
+          ? 'open file'
+          : 'open resource in browser'
+      }
+      placement="right"
       onClick={(): void => {
         window.ipcRenderer.invoke(IpcChannel.OpenLink, openLinkArgs);
       }}
+      className={!openLinkArgs.link ? classes.hidden : undefined}
```
[the remark this answers is not in the document]
Comment (mayayunx) [reply src/Frontend/Components/GoToLinkButton/GoToLinkButton.tsx:90]: But then it needs to be packed into clsx right? And Leslie commented above that he finds it unclean to use boolean as args in clsx.

[the remark this answers is not in the document]
Comment (benedikt-richter) [reply src/Frontend/Components/GoToLinkButton/GoToLinkButton.tsx:90]: Ok. is not required

**Review by leslielazzarino — APPROVED** [review]

**Pull request merged**